src/commonMain/kotlin/dev/evo/elasticmagic/DocSource.kt

```diff
@@ -234,12 +240,33 @@ open class DocSource : BaseDocSource() {
         val type: FieldType<V>,
         val isRequired: Boolean,
     ) {
+        constructor(
+            name: String,
+            type: FieldType<V>,
+            isRequired: Boolean,
+            defaultValue: () -> V
+        ) : this(name, type, isRequired) {
+            _defaultValue = defaultValue
+            hasDefault = true
+        }
+
         var isInitialized: Boolean = false
             private set
+        var preInitialized: Boolean = false
+            private set
+        var hasDefault: Boolean = false
+            private set
 
         private var _value: V? = null
+        private var _defaultValue: () -> V? = { null }
         var value: V?
-            get() = _value
+            get() {
+                if (!isInitialized && hasDefault && !preInitialized) {
+                    _value = _defaultValue()
+                    preInitialized = true
```

Comment:
Why couldn't we just use `isInitialized` flag?

Reply:
You are right. I overthinked. Fixed

Reply:
I just thought in some cases it would be useful to know if field was initialized with value from elastic result or by user and not with default value. What do you think?

Reply:
Cannot think of a case where we should distinguish default value. When we really need it we'll implement it.